Homework24/src/Client.java

```diff
@@ -0,0 +1,157 @@
+import java.awt.BorderLayout;
+import java.awt.Desktop;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JTextArea;
+import javax.swing.JTextField;
+
+public class Client extends JFrame {
+
+	private static final long serialVersionUID = 6463534731269380864L;
+
+	Socket client = null;
+	BufferedWriter bw = null;
+	BufferedReader br = null;
+	String path = "";
+	JButton send = new JButton("Send !");
+	JTextArea textArea = new JTextArea();
+	JTextField textField = new JTextField();
+
+	public Client() {
+		send.addActionListener(new ActionListener() {
+
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				try {
+
+					bw.write(textField.getText());
+					bw.newLine();
+					bw.flush();
+					if (textField.getText().length() > 0) {
```

Comment:
Ovaj if blok bi trebao obuhvatiti i ove 3 linije koda iznad. Prema trenutacnom kodu iako korisnik nije nista unio (vec samo pritisnuo send) poslat ce se empty string. To se svejedno nece prikazati na klijentu obzirom da postoji check koji provjerava da li je duzina dobijenog stringa veca od 0. Svjedno, trebali bismo zabraniti slanje bilo cega ukoliko je ovaj input field prazan.